src/test/java/io/vavr/beanvalidation2/valueextraction/OptionTest.java

```diff
@@ -0,0 +1,109 @@
+/*  __    __  __  __    __  ___
+ * \  \  /  /    \  \  /  /  __/
+ *  \  \/  /  /\  \  \/  /  /
+ *   \____/__/  \__\____/__/
+ *
+ * Copyright 2014-2019 Vavr, http://vavr.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vavr.beanvalidation2.valueextraction;
+
+import io.vavr.beanvalidation2.ValidatorSupplier;
+import io.vavr.collection.HashSet;
+import io.vavr.collection.List;
+import io.vavr.collection.Set;
+import io.vavr.control.Option;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ElementKind;
+import javax.validation.Path;
+import javax.validation.Validator;
+import javax.validation.constraints.NotBlank;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class OptionTest {
+	private Validator validator;
+
+	@Before
+	public void setUp() {
+		this.validator = ValidatorSupplier.INSTANCE.get();
+	}
+
+	@Test
+	public void testOnNone() {
+		// given
+		TestBean subject = new TestBean();
+
+		// when
+		Set<ConstraintViolation<TestBean>> violations =
+				HashSet.ofAll(validator.validate(subject));
+
+		// then
+		assertThat(violations).hasSize(0);
+	}
+
+	@Test
+	public void testOnValid() {
+		// given
+		TestBean subject = new TestBean();
+		subject.set("valid");
+
+		// when
+		Set<ConstraintViolation<TestBean>> violations =
+				HashSet.ofAll(validator.validate(subject));
+
+		// then
+		assertThat(violations).hasSize(0);
+	}
+
+	@Test
+	@Ignore("On line 79 value is <iterable element> instead of <element>")
```

Comment:
This should be fixed in some way

Reply:
see discussion below (comment on 2019-08-03)